### PR TITLE
Serve both components by `gin-gonic`

### PR DESCRIPTION
**Pull request by VitaliLupusor**
# Description
Both the frontend and the API are now served by the `gin` framework.

**Important!**
When developing locally, make sure to move the `dist` folder created by the `npm run build` command into the `./api` directory.

### CODE REVIEW

File: api/main.go [hunk 1]
```diff
@@ -22,6 +32,48 @@ func main() {
...
 	}
 
-	address := fmt.Sprintf(":%d", PORT)
-	router.Run(address)
+	return router
+}
+
+func makeFrontendRouter() http.Handler {
+	router := gin.Default()
+	defaultCors := cors.Default()
```
Comment (BarryFoye) [inline api/main.go:40]: Don't think this is needed. Should be able to remove it.

File: api/main.go [hunk 2]
```diff
@@ -22,6 +32,48 @@ func main() {
...
+
+	// Mount middleware
+	router.Use(defaultCors)
+
+	router.StaticFS("/", gin.Dir(FRONTEND_DIR, true))
```
Comment (BarryFoye) [inline api/main.go:45]: not essential but worth taking a look at https://github.com/gin-contrib/static. Might be useful for symplifying what we have but may not.

File: app/package.json [hunk 3]
```diff
@@ -5,6 +5,7 @@
   "scripts": {
     "serve": "vue-cli-service serve",
     "build": "vue-cli-service build",
+    "build-dev": "vue-cli-service build --dest ../api/dist",
```
Comment (BarryFoye) [inline app/package.json:8]: This tells the builder to place the dist folder in the api folder for dev purposes.

File: app/src/components/HolidayList.vue [hunk 4]
```diff
@@ -13,8 +13,24 @@
...
+  IonList,
+  IonListHeader,
+  IonRow,
+  IonCard,
+  IonCardTitle,
+  IonCardContent
+} from '@ionic/vue'
 
 export default defineComponent({
+  components: {
```
Comment (BarryFoye) [inline app/src/components/HolidayList.vue:26]: The issue was that the imports and components weren't there so no wonder it struggled. I think serving with the ionic plugin probably had the node_modules in context so could resolve the components on the fly but the build never had them packaged.

File: app/src/router/index.ts [hunk 5]
```diff
@@ -1,22 +1,12 @@
...
-    path: '/home',
     name: 'Home',
-    component: HomePage
-  },
-  {
-    path: '/hello',
-    name: 'Hello',
-    component: () => import('@/views/HomePage.vue'),
-  },
+    component: () => import('@/views/HomePage.vue')
```
Comment (BarryFoye) [inline app/src/router/index.ts:8]: This solves the routing issue where 404 would come back on refresh.

**Review by BarryFoye — APPROVED** [review]

File: app/package.json (hunk 3 above)
Comment (VitaliLupusor) [reply app/package.json:8]: Well done.

**Review by VitaliLupusor — COMMENTED** [review]
I am ready to merge. Let me know, if you think there are changes necessary

**Review by BarryFoye — APPROVED** [review]

**Pull request merged**